### PR TITLE
Fix the way to get "quantization_config" from model's coniguration

### PR DESCRIPTION
**Pull request by lvhan028**
after #2090

### CODE REVIEW

File: lmdeploy/turbomind/deploy/converter.py [hunk 1]
```diff
@@ -161,17 +161,77 @@ def pack_model_repository(workspace_path: str):
                dst=osp.join(model_repo_dir, 'postprocessing'))
 
 
+def find_quantization_config(nested, target_key):
```
Comment (lzhangzz) [inline lmdeploy/turbomind/deploy/converter.py:164]: Will this recurse infinitely if the config back links to itself?

File: lmdeploy/turbomind/deploy/converter.py [hunk 2]
```diff
@@ -161,17 +161,77 @@ def pack_model_repository(workspace_path: str):
...
+                f'unsupported quant config: {quant_config}'
+        elif quant_method == 'gptq':
+            assert not quant_config.get('desc_act', False) and \
+                quant_config.get('sym', True), \
+                f'unsupported quant config: {quant_config}'
+        else:
+            assert 0, f'unsupported quant_config: {quant_config}'
+
+    if engine_config.model_format in ['awq', 'gptq']:
+        assert group_size, \
```
Comment (lzhangzz) [inline lmdeploy/turbomind/deploy/converter.py:232]: Need to assert group_size == 128

**Comment by zhulinJulia24** [comment]
I get error like

File: lmdeploy/turbomind/deploy/converter.py (hunk 1 above)
Comment (lvhan028) [reply lmdeploy/turbomind/deploy/converter.py:164]: The recursive procedure happens only when the `nested` is a dict or a list. 
I don't think there is a case that the config links back to itself given it is a dict or a list.

**Comment by lvhan028** [comment]
> Get error like when use OpenGVLab/InternVL2-40B-inner-4bits with tp = 2
> 
> ```
> 
> The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.
> You are using a model of type internvl_chat to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
> The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.
> You are using a model of type internvl_chat to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
> Process Process-3:
> Traceback (most recent call last):
>   File "/usr/lib/python3.10/multiprocessing/process.py", line 314, in _bootstrap
>     self.run()
>   File "/usr/lib/python3.10/multiprocessing/process.py", line 108, in run
>     self._target(*self._args, **self._kwargs)
>   File "/__w/lmdeploy/lmdeploy/autotest/utils/pipeline_chat.py", line 298, in run_pipeline_vl_chat_test
>     pipe = pipeline(hf_path, backend_config=backend_config)
>   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/api.py", line 72, in pipeline
>     backend_config = autoget_backend_config(model_path, backend_config)
>   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/archs.py", line 89, in autoget_backend_config
>     backend = autoget_backend(model_path)
>   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/archs.py", line 42, in autoget_backend
>     turbomind_has = is_supported_turbomind(model_path)
>   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/turbomind/supported_models.py", line 99, in is_supported
>     support_by_turbomind = _is_head_dim_128(cfg.llm_config)
>   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/turbomind/supported_models.py", line 67, in _is_head_dim_128
>     num_attn_head = cfg.num_attention_heads
> AttributeError: 'dict' object has no attribute 'num_attention_heads'
> ```

I didn't reproduce it. Please check `config.json` at your side

**Comment by zhulinJulia24** [comment]
> > Get error like when use OpenGVLab/InternVL2-40B-inner-4bits with tp = 2
> > ```
> > 
> > The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.
> > You are using a model of type internvl_chat to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
> > The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.
> > You are using a model of type internvl_chat to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
> > Process Process-3:
> > Traceback (most recent call last):
> >   File "/usr/lib/python3.10/multiprocessing/process.py", line 314, in _bootstrap
> >     self.run()
> >   File "/usr/lib/python3.10/multiprocessing/process.py", line 108, in run
> >     self._target(*self._args, **self._kwargs)
> >   File "/__w/lmdeploy/lmdeploy/autotest/utils/pipeline_chat.py", line 298, in run_pipeline_vl_chat_test
> >     pipe = pipeline(hf_path, backend_config=backend_config)
> >   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/api.py", line 72, in pipeline
> >     backend_config = autoget_backend_config(model_path, backend_config)
> >   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/archs.py", line 89, in autoget_backend_config
> >     backend = autoget_backend(model_path)
> >   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/archs.py", line 42, in autoget_backend
> >     turbomind_has = is_supported_turbomind(model_path)
> >   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/turbomind/supported_models.py", line 99, in is_supported
> >     support_by_turbomind = _is_head_dim_128(cfg.llm_config)
> >   File "/usr/local/lib/python3.10/dist-packages/lmdeploy/turbomind/supported_models.py", line 67, in _is_head_dim_128
> >     num_attn_head = cfg.num_attention_heads
> > AttributeError: 'dict' object has no attribute 'num_attention_heads'
> > ```
> 
> I didn't reproduce it. Please check `config.json` at your side

It seems related to transformers version. I can reproduce it on transformers = 4.41.2.

**Review by lzhangzz — APPROVED** [review]

**Pull request merged**